lib/more_core_extensions/core_ext/object/descendants.rb

```diff
@@ -0,0 +1,17 @@
+require 'active_support/core_ext/class/subclasses'
+
+module MoreCoreExtensions
+  module Descendants
+    #
+    # Retrieve a descendant by its name
+    #
+    def descendant_get(desc_name)
+      return self if desc_name == to_s || desc_name.nil?
+      klass = descendants.find { |desc| desc.name == desc_name }
+      raise "#{desc_name} is not a descendant of #{name}" unless klass
```

Comment:
Although const_get returns NameError (making it consistent), that error is reserved for when the Name doesn't exist...this exists, but isn't valid.  I'm not against NameError, just thinking out loud...maybe ArgumentError is better?